src/debonair/Restaurant.java

```diff
@@ -0,0 +1,137 @@
+// user defined package
+package debonair;
+
+import java.time.LocalDateTime;
+// imports from in-built packages
+import java.util.Iterator;
+import java.util.Scanner;
+
+// importing classes from other packages
+import resources.Constants;
+
+// Restaurant class
+public class Restaurant implements IRestaurant {
+	/**
+	 * This method is used to show menu to the customer
+	 */
+	public void showMenu() {
+		System.out.println("The menu has following items:-");
+		// Iterator created to print items from Menu class's itemMenu List
+		Iterator<Item> itemsIterator = Menu.ITEM_MENU.iterator();
+		while (itemsIterator.hasNext()) {
+			Item item = itemsIterator.next();
+			System.out.println(item.toString());
+		}
+	}
+
+	/**
+	 * This method will be used to take order from the customer
+	 */
+	public void takeOrder() {
+		Order.createOrder();
+	}
+
+	/**
+	 * This method will be used to check customer's order status
+	 * 
+	 * @param orderID
+	 */
+	public boolean checkStatus(int orderId) {
+		// not so important as no functionality in assignment of collections so just
+		// returning true
```

Comment:
Create a collection with name orders .save all orders to this collection.
Add attriblutes to item class preperationTimeSingleUnit and quantityOrdered.
Add an attribute to order class total time which would be a sum of preperation time of all items in the order.
whenever status is checked it should calculate the remaining time. and display the status.if time has passed display order already delivered.
also think and handle other corner scenarios.